test/Microsoft.AspNetCore.Http.Tests/Internal/ApplicationBuilderTests.cs

```diff
@@ -31,5 +33,46 @@ public void PropertiesDictionaryIsDistinctAfterNew()
 
             Assert.Equal("value1", builder1.Properties["test"]);
         }
+
+        [Fact]
+        public void BuildWithEndpointReturnsEndpointRequestDelegate()
+        {
+            var endpointInvoked = false;
+            var endpoint = new TestEndpoint(
+                context =>
+                {
+                    endpointInvoked = true;
+                    return Task.CompletedTask;
+                },
+                EndpointMetadataCollection.Empty,
+                string.Empty);
+
+            var featureCollection = new FeatureCollection();
+            featureCollection.Set<IEndpointFeature>(new TestEndpointFeature
+            {
+                Endpoint = endpoint
+            });
+            var builder = new ApplicationBuilder(null, featureCollection);
+            var app = builder.Build();
+
+            var httpContext = new DefaultHttpContext();
+
+            app.Invoke(httpContext);
+
+            Assert.True(endpointInvoked);
+        }
+
+        private class TestEndpoint : Endpoint
```

Comment:
This makes me wonder why we don't just make Endpoint non abstract. It'll remove the need for this.

Reply:
Now that Endpoint has RequestDelegate on it it does feel a lot less abstract. What do you think @rynowak ?

Reply:
I think it would be fine now that the request delegate has moved down.